mailpoet/assets/js/src/automation/integrations/mailpoet/steps/someone-subscribes/edit/list_panel.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useState } from 'react';
+import { PanelBody, Spinner } from '@wordpress/components';
+import { dispatch, useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { uniq } from 'lodash';
+import { storeName } from '../../../../../editor/store';
+import { MailPoetAjax } from '../../../../../../ajax';
+import { RawSegment, Segment } from './segment';
+import { PlainBodyTitle } from '../../../../../editor/components/panel';
+import {
+  FormTokenItem,
+  FormTokenField,
+} from '../../../components/form-token-field';
+
+export function ListPanel(): JSX.Element {
+  const { selectedStep } = useSelect(
+    (select) => ({
+      selectedStep: select(storeName).getSelectedStep(),
+    }),
+    [],
+  );
+
+  const [segments, setSegments] = useState<FormTokenItem[]>([]);
+
+  const anyValue = {
+    id: 0,
+    name: __('Any list', 'mailpoet'),
+  };
+
+  useEffect(() => {
+    const getData = async () => {
+      const data = await MailPoetAjax.post({
```

Comment:
Could we make this non-async, without the loaders in the sidebar? Or is that not so easy? Not sure what will be the performance impacts if we make all steps sync 🤔

Reply:
Addressed in [c04a306](https://github.com/mailpoet/mailpoet/pull/4371/commits/c04a306efb701df4ca62d8ef2b1db841a3211b28)